src/sema/ast.rs

```diff
@@ -59,6 +59,10 @@ pub enum Type {
     Slice,
     /// We could not resolve this type
     Unresolved,
+    /// When we advance a pointer, it cannot be any of the previous types.
+    /// e.g. Type::Bytes is a pointer to struct.vector. When we advance it, it is a pointer
+    /// to latter's data region
+    BufferPointer,
```

Comment:
Our CFG representation has a severe limitation: pointers aren't represented. This makes codegen very bad in some situations. :/